packages/loot-core/src/server/accounts/transfer.js

```diff
@@ -101,12 +125,24 @@ export async function removeTransfer(transaction) {
       // would invalidate the whole split transaction. Instead of turn
       // it into a normal transaction
       await db.updateTransaction({
-        id: transaction.transfer_id,
+        id: transferTrans.id,
         transfer_id: null,
         payee: null,
       });
+    } else if (transferTrans.cleared) {
```

Comment:
Also, do you think it would make sense to combine this logic with the child logic above so that child transactions also benefit from getting the payee updated? (or does that not make sense)